org.eclipse.wildwebdeveloper/src/org/eclipse/wildwebdeveloper/debug/node/NodeRunDAPDebugDelegate.java

```diff
@@ -99,4 +110,124 @@ public void launch(ILaunchConfiguration configuration, String mode, ILaunch laun
+		String program = (String)param.get(PROGRAM);
+		String cwd = (String)param.get(CWD);
+		
+		if (program != null && program.endsWith(".ts")) {
+			// TypeScript Source Mappings Configuration
+			String tsConfigPath = cwd + "/tsconfig.json";
+			
+			String errorMessage = null;
+			Map<String, Object> tsConfig = readTsConfig(tsConfigPath);
+			System.out.println("CompilerOptions: " +  tsConfig.toString());
+
+			Map<String, Object> co = (Map<String, Object>)tsConfig.get("compilerOptions");
+			if (tsConfig.isEmpty() || co == null) {
+				errorMessage = Messages.NodeDebug_TSConfirError_NoTsConfig;
+				co = new HashMap<>();
+			}
+
+			//TS Compiler Options
+			param.putAll(co);
+
+			if (errorMessage == null) {
+				Object option = co.get("sourceMap");
+				boolean sourceMap  = option instanceof Boolean ? ((Boolean)option).booleanValue() : false;
+				if (!sourceMap) {
+					errorMessage = Messages.NodeDebug_TSConfirError_SourceMapIsNotEnabled;
+				}
+			}
+
+			// Override "outDir" option by converting it to an absolute path
+			boolean outDirOrFileIsSet = false;
+			Object option = co.get("module");
+			String module = option instanceof String ? ((String)option).trim() : null;
+						
+			option = co.get("outDir");
+			String outDir = option instanceof String ? ((String)option).trim() : null;
+			if (outDir != null && outDir.length() > 0 && !".".equals(outDir) && !"./".equals(outDir)) {
+				param.put("outDir", cwd + "/" + outDir);
+				outDirOrFileIsSet = true;
+			}
+			
+			option = co.get("outFile");
+			String outFile = option instanceof String ? ((String)option).trim() : null;
+			if (outFile != null && outFile.length() != 0) {
+				param.put("outFile", cwd + "/" + outFile);
+				outDirOrFileIsSet = true;
+				
+				if (!"amd".equalsIgnoreCase(module)  && !"system".equalsIgnoreCase(module)) {
+					errorMessage = Messages.NodeDebug_TSConfigError_OutDirNotSupportedModule;
+				}
+			}
+			
+			if (!outDirOrFileIsSet && errorMessage == null) {
+				errorMessage = Messages.NodeDebug_TSConfigError_OutDirIsNotSet;
+			}
+
+			if (errorMessage != null) {
+				// Display error message
+				final int[] result = new int[1];
+				final String dialogMessage = errorMessage;
+				
+				Display.getDefault().syncExec(new Runnable() {
+						@Override
+						public void run() {
+							MessageDialog dialog = new MessageDialog(DebugUIPlugin.getShell(),
+									Messages.NodeDebug_TSConfirError_Title, null, dialogMessage, MessageDialog.QUESTION_WITH_CANCEL,
+									2, Messages.NodeDebug_TSConfirError_OpenTSConfigInEditor,
+									Messages.NodeDebug_TSConfirError_StartDebuggingAsIs, Messages.NodeDebug_TSConfirError_Cancel);
+							result[0] = dialog.open();
+						}
+					});
+				
+				if (result[0] == 0) {
+					// Open TSConfig in editor
+					Display.getDefault().asyncExec(new Runnable() {
+						@Override
+						public void run() {
+							try {
+								IDE.openEditor(
```

Comment:
Why explicitly opening the editor? I don't think launch usually do that (ie JDT doesn't enforce opening the editor to run).

Reply:
I can remove that `Open "tsconfig.json" in Editor` button. It's just was quite comfortable to use it to test/change different TSConfig settings

Reply:
Ah OK, it's only in case of faulty tsconfig.json, it's indeed good as it. Let's keep it.